### ISSUE TITLE
Travis CI: build logs are truncated

### ISSUE THREAD

**Issue by php-coder** [php-coder/mystamps#631]
Some time ago I noticed that some build logs are truncated. First it was always happened when I tried to show the results of the integration tests. But from some point, it has started to happen always on all branches.

The issue that I created: https://github.com/travis-ci/travis-ci/issues/8189

**Comment by php-coder** [comment]
> First it was always happened when I tried to show the results of the integration tests.

Here are workarounds that I added:
- gzip the report before encoding: 921974d9f9092fd4395479811f501b041c234d18
- dump XML report instead of HTML: a3090467aee4843137459bad072df23d0e0b74c4

> But from some point, it has started to happen always on all branches.

In my first attempt to fix it, I tried to show the output from the script as fast as possible: 525bf72ba279a149f3f9fb0a5acc146de6b64f3d I thought that there is some time out for waiting for a script output. But it didn't help.

**Comment by php-coder** [comment]
In order to fix this issue we need to split `check-build-and-verify.sh` script into separate pieces. If we do that, the output from the different scripts will be accounted separately and we won't exceed the output buffer size.

**Comment by php-coder** [comment]
Unfortunately it didn't help for the case when build was failing.

I found another issue that looks like mine but with more realistic theory of what is happening: https://github.com/travis-ci/travis-ci/issues/6018

So, we need to put `sleep` after script execution, so Travis will be able to capture script output before it will exit.

I played with it and it didn't fully solve this issue. It turned out that it's important to not use `exit` in `script` block because it leads to exiting from the the whole build and `after_failure` never gets executed.

But I didn't use `exit` and `after_failure` still wasn't get executed. It has happened because I activated `errexit` option in the `before_script` block that was inherited by the `script` block.
